mealie/routes/validators/validators.py

```diff
@@ -0,0 +1,45 @@
+from fastapi import APIRouter, Depends
+from typing import Union
+from sqlalchemy.orm.session import Session
+
+from mealie.db.db_setup import generate_session
+from mealie.repos.all_repositories import get_repositories
+from mealie.schema.user import UserBase
+from mealie.schema.response import ErrorResponse, SuccessResponse
+from mealie.db.models.users import User
+from mealie.db.models.group import Group
+from mealie.db.models.recipe.recipe import RecipeModel
+
+
+router = APIRouter()
+
+
+def create_response(value, type: str):
+    if value is None:
+        return SuccessResponse.respond(message=f"No {type} exists")
+    return ErrorResponse.respond(message=f"{type.capitalize()} already exists")
+
+
+@router.get("/user/{name}", response_model=Union[ErrorResponse, SuccessResponse])
+async def validate_user(name: str, session: Session = Depends(generate_session)):
```

Comment:
Since these methods aren't async they shouldn't be declared as async. FastAPI will execute async methods on the main thread and these will all block. If you remove the async they'll be thrown into a threadpool which will yield better performance.